lib/Command/FixEncryptedVersion.php

```diff
@@ -0,0 +1,204 @@
+			$directoryContent = $this->view->getDirectoryContent($root);
+			foreach ($directoryContent as $file) {
+				$path = $root . '/' . $file['name'];
+				if ($this->view->is_dir($path)) {
+					$directories[] = $path;
+				} else {
+					$output->writeln("Verifying the content of file $path");
+					$this->verifyFileContent($path, $output);
+				}
+			}
+		}
+	}
+
+	/**
+	 * @param string $path
+	 * @param OutputInterface $output
+	 * @param bool $allowOnce
+	 */
+	private function verifyFileContent($path, OutputInterface $output, $allowOnce = true) {
+		try {
+			if ($this->view->readfilePart($path, 0, 9000) !== false) {
+				$output->writeln("<info>The file $path is: OK</info>");
+			}
+			return true;
+		} catch (HintException $e) {
+			\OC::$server->getLogger()->warning("Issue: " . $e->getMessage());
+			//If allowOnce is set to false, this becomes recursive.
+			if ($allowOnce === true) {
+				//Lets rectify the file by correcting encrypted version
+				return $this->correctEncryptedVersion($path, $output);
+			}
+			return false;
+		}
+	}
+
+	/**
+	 * @param string $path
+	 * @param OutputInterface $output
+	 */
+	private function correctEncryptedVersion($path, OutputInterface $output) {
+		$skipDecrymentedVersion = null;
+
+		$fixedFileInfo = ['filename' => '', 'encryptedVersion' => 0];
+
+		$fileInfo = $this->view->getFileInfo($path);
+		$fileId = $fileInfo->getId();
+		$encryptedVersion = $fileInfo->getEncryptedVersion();
+		$oldEncryptedVersion = $encryptedVersion;
+
+		$storage = $fileInfo->getStorage();
+
+		$cache = new Cache($storage);
+		$fileCache = $cache->get($fileId);
+		$fileCacheStorage = $fileCache->getData()['storage'];
+
+		if (\strpos($fileCacheStorage, "shared::") !== false) {
+			$output->writeln("<info>The file: $path is a share. Hence kindly fix this by running the script under the owner of share</info>");
+			return true;
+		}
+
+		if ($encryptedVersion >= 0) {
+			if ($encryptedVersion === 0) {
+				$skipDecrymentedVersion = true;
+			}
+			//test by decrementing the value till 1 and if nothing works try incrementing
+			if (($skipDecrymentedVersion !== true) && ($fixedFileInfo['filename'] !== $path)) {
+				$encryptedVersion--;
+				while ($encryptedVersion > 0) {
+					if ($fixedFileInfo['filename'] === $path && $fixedFileInfo['encryptedVersion'] > 0) {
+						return true;
+					}
+
+					$cacheInfo = ['encryptedVersion' => $encryptedVersion, 'encrypted' => $encryptedVersion];
+					$cache->put($fileCache->getPath(), $cacheInfo);
+					$output->writeln("<info>Decrement the encrypted version to $encryptedVersion</info>");
+					if ($this->verifyFileContent($path, $output, false) === true) {
+						$output->writeln("<info>Fixed the file $path with version " . $encryptedVersion . "</info>");
+						return true;
+					}
+					$encryptedVersion--;
+				}
+			}
+
+			//So decrementing did not worked. Now lets increment. Max increment is till 25
+			$increment = 1;
+			while ($increment <= 25) {
```

Comment:
not sure if we need to go that far, this would also take more time to process for many files

I suspect that the root cause would only cause a discrepancy of +1.
maybe use 5 here just to be safe.

Reply:
Done. Fixed with 5.